src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1317,18 +1317,17 @@ private boolean confirmClose(BasePanel panel) {
             // The user wants to save.
             try {
                 SaveDatabaseAction saveAction = new SaveDatabaseAction(panel, Globals.prefs);
-                if (!saveAction.save()) {
-                    // The action was either canceled or unsuccessful.
-                    output(Localization.lang("Unable to save library"));
-                    return false;
+                if (saveAction.save()) {
+                    // Saved, now exit.
+                    return true;
                 }
+                // The action was either canceled or unsuccessful.
+                output(Localization.lang("Unable to save library"));
             } catch (Throwable ex) {
-                return false;
+                LOGGER.error("A problem occurred when saving the file", ex);
```

Comment:
Please also add a `dialogService.showError()` call, because we also want to notify the user that we were unable to save the library.